### PR TITLE
Fix tutorial about shuffing before sharding

### LINKED ISSUE #709: Update tutorial about shuffling before sharding
### 📚 The doc issue

The [tutorial](https://pytorch.org/data/beta/tutorial.html#working-with-dataloader) needs to update the actual reason about shuffling before sharding. It's not accurate.
Shuffling before sharding is required to achieve global shuffling rather than only shuffling inside each shard.

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by ejguan**
Fixes #709

Per title

### CODE REVIEW

**Review by NivekT — APPROVED** [review]
LGTM!

Do you also want to mention the exception for internal use case here? Something about data being sharded prior to loading? (I'm probably poorly paraphrasing)
